### PR TITLE
Server side Authentication

### PR DESCRIPTION
**Pull request by kofimeighan**

### CODE REVIEW

File: portfolio/src/main/webapp/leaveamessage.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <!--Navigation bar at the top of the screen-->
+    <ul class="nav">
+      <li><a href="index.html">Home</a></li>
+      <li><a href="education.html">Education & Activities</a></li>
+      <li><a href="pictures.html">Film Pictures</a></li>
+      <li><a href="mixes.html">Music Mixes</a></li>
+      <li><a href="leaveamessage.html">Leave me a message!</a></li>
+    </ul>
+    
+    	<body align="center" onload="checkLoginStatus(); getLimitedMessage();">
```
Comment (kennzoid) [inline portfolio/src/main/webapp/leaveamessage.html:32]: With just two functions this is probably fine but it won't scale as you add many functions.

Prefer something like onload="initializePage()" or something like that and call checkLoginStatus() and getLimitedMessage() within that.

File: portfolio/src/main/webapp/leaveamessage.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <h5 style="font-size:30px"><b>Leave me a message!</b></h5>
```
Comment (kennzoid) [inline portfolio/src/main/webapp/leaveamessage.html:33]: Style should stay in CSS.

https://google.github.io/styleguide/htmlcssguide.html#Separation_of_Concerns

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -13,15 +13,12 @@
...
-  let amountOfComments = document.getElementById("commentAmount").value;
-  ?amountOfComments='+ amountOfComments);*/
-
-  const response = await fetch('/data');
+	let amountOfComments = document.getElementById("commentAmount").value
+  const response = await fetch('/data?amtOfComments='+amountOfComments);
   const messages = await response.json();
   document.getElementById('message-container').innerText = '';
 
+	document.getElementById('data-page-message-container').innerHTML= '';
```
Comment (kennzoid) [inline portfolio/src/main/webapp/script.js:21]: nit: fix indentation

File: portfolio/src/main/java/com/google/sps/servlets/UsernameServlet.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+public class UsernameServlet extends HttpServlet {
+
+  public static final String USERNAME = "username";
+  public static final String TABLE_NAME = "Messages";
+  public static final String ID = "id";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      UserService userService = UserServiceFactory.getUserService();
+      //TO-DO:
```
Comment (kennzoid) [inline portfolio/src/main/java/com/google/sps/servlets/UsernameServlet.java:41]: nit: missing todo (:

File: portfolio/src/main/webapp/username.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      <li><a href="index.html">Home</a></li>
+      <li><a href="education.html">Education & Activities</a></li>
+      <li><a href="pictures.html">Film Pictures</a></li>
+      <li><a href="mixes.html">Music Mixes</a></li>
+    </ul>
+
+    <h3>Set your nickname:</h3>
+      <body align="center">
+        <form method="POST" action="/username">
+        <input name="username" id="username" /></br></br>
```
Comment (kennzoid) [inline portfolio/src/main/webapp/username.html:34]: Both name and id are the same for now?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -89,25 +92,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
         //printing each individual message
         for(int i=0; i< messages.size(); i++) {
             response.getWriter().println(messages.get(i));
         }
-        //TODO:
-        //look at a better way to clean the boxes
-        //get the input # from html and send to JS without redirecting so that multiple users
-        //can use the site at once
-        response.sendRedirect("/index.html");
+        response.sendRedirect("/leaveamessage.html");
```
Comment (kennzoid) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:105]: Messages is only going to have 0 or 1 elements. Is that intended?

File: portfolio/src/main/webapp/leaveamessage.html (hunk 1 above)
Comment (kofimeighan) [reply portfolio/src/main/webapp/leaveamessage.html:32]: Moved both functions into an onLoad() function

File: portfolio/src/main/webapp/leaveamessage.html (hunk 2 above)
Comment (kofimeighan) [reply portfolio/src/main/webapp/leaveamessage.html:33]: Fixed

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (kofimeighan) [reply portfolio/src/main/webapp/script.js:21]: Done

File: portfolio/src/main/java/com/google/sps/servlets/UsernameServlet.java (hunk 4 above)
Comment (kofimeighan) [reply portfolio/src/main/java/com/google/sps/servlets/UsernameServlet.java:41]: I'm not realizing I don't use this servlet. I'll go ahead and delete it

Comment (kofimeighan) [reply portfolio/src/main/java/com/google/sps/servlets/UsernameServlet.java:41]: now*

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (kofimeighan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:105]: No, I have fixed it to write to the client only if the message has content.

**Pull request merged**